### DISCUSSION TITLE
Best way to paginate ?

### DISCUSSION THREAD

**Discussion by Beludia** [sqlpage/SQLPage#39] [Q&A]
Hello, I have a table with 8000 entry and growing, It make the website a little slow to show everything, what would be the best way to paginate ? thanks !

**Comment by lovasoa** [comment]
Hey ! Can you simple add an `ORDER BY` and a `LIMIT` to your query ?

**Comment by Beludia** [comment]
Yes but how to I navigate from id 1 > 50, 50 > 100 ? I guess I have to use OFFSET, but how do I add page link ? Is theire a way to do it automaticaly or will I have to add offset link myself everytime ? 
I already added LIMIT 500 but since I want to search in the entire table the search bar only look in the 500 results shown.
Thanks for the help !

**Reply by lovasoa** [reply]
Unfortunately, we don't have something built-in for that, but you can build that yourself with the `form` component for search and the steps component for navigation between pages.

https://sql.ophir.dev/documentation.sql?component=steps#component


However, I'm not sure a long table with thousands of elements is very useful for navigation. 

Maybe you should let the users filter data along facets with the list component?

**Reply by lovasoa** [reply]
What does the large table contain?

**Reply by Beludia** [reply]
a list of names of characters from an mmo, and the guild they are in, I thought about steps, i'll have to try it thanks.

**Reply by lovasoa** [reply]
Then maybe make the main page a list of guilds, or a list of first letters from the name, and show the filtered list of players only on a second page ?

**Reply by Beludia** [reply]
that seems a good idea the letters thanks !

**Reply by lovasoa** [reply]
And if I may: are the application you are building and the dataset pubic? 8000 rows is large but not huge, and SQLPage streams the data directly to the browser, so it worries me a little that it becomes unusably slow at only 8000 rows.

**Reply by Beludia** [reply]
Its not open actually but I wouldnt mind open it for you to check, but I wouldnt say its unusably slow, it's just not really useful for me to show everything, but having the search instant is better

**Comment by lovasoa** [comment]
I answered the question more completely, with an example, here: https://github.com/lovasoa/SQLpage/issues/84

**Reply by Beludia** [reply]
Thanks, but as you said in the post, pagination would still require to load the entire page and I need that to keep to fast searching ! But thanks for thinking of it !
